[Daphne789] ip

Daphne789 opened the pull request:
# EchoBot
> “Your mind is for having ideas, not holding them.” – David Allen
> [(source)](https://dansilvestre.com/productivity-quotes).

### EchoBot frees your mind of having to remember things you need to do. 
### It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

### All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smiley:

And it is **FREE**!

### Features:

- [x] Managing tasks
- [x] Managing deadlines (:soon:)
- [x] Reminders (:soon:)

### If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Daphne789 committed:
Completed Level-0

Daphne789 committed:
Modified to ignore .class files

Daphne789 committed:
Completed Level-1

Daphne789 committed:
Completed Level-2

Daphne789 committed:
Completed Level-3

Daphne789 committed:
Completed Level-4

Daphne789 committed:
Completed Automated Text UI Testing

Daphne789 committed:
Completed Level-5

Daphne789 committed:
Completed Level-6

Daphne789 committed:
Separate class into a new file

Daphne789 committed:
Remove some classes

Daphne789 committed:
Add save feature

Daphne789 committed:
Complete Level-8

Daphne789 committed:
Complete Level-7

Daphne789 committed:
Add data folder

Daphne789 committed:
Complete Level-8

Daphne789 committed:
Fix ArrayIndexOutOfBoundsException

Daphne789 committed:
Complete A-MoreOOP

Daphne789 committed:
Complete A-Packages

Daphne789 committed:
Merge branch 'add-gradle-support'

Daphne789 committed:
Complete A-Gradle

Daphne789 committed:
Complete A-JUnit

Daphne789 committed:
Add isTaskDone method

Daphne789 committed:
Complete A-Jar

Daphne789 committed:
Add JavaDoc

Daphne789 committed:
Complete A-CodingStandard

Daphne789 committed:
Complete Level-9

Daphne789 committed:
Add JavaDoc

Daphne789 committed:
Complete A-CheckStyle

Daphne789 committed:
Complete Level-10

ruth-lim commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
Perhaps you can change the variable name to something like "dueDate" so that it is more intuitive.

ruth-lim commented on src/main/java/EchoBot.java:
```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
```
Perhaps you want to move your EchoBot class under the duke package so that all classes are minimally under the duke package 😄

ruth-lim commented on src/main/java/EchoBot.java:
```diff
@@ -0,0 +1,92 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.ui.Ui;
```
I like how you import the classes individually! 👍

ruth-lim replied:
Maybe you can consider renaming your package to `echobot` instead of `duke` so that it is standardised?

ruth-lim reviewed (commented):
In general, I think you have done a great job naming all the variables, methods and classes. It is very intuitive for the reader. Just some nits to fix. Well done! 👍

newway1814 commented on src/main/java/EchoBot.java:
```diff
@@ -0,0 +1,154 @@
+     */
+    public EchoBot() {
+        frame = new JFrame("EchoBot");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```
1st

newway1814 commented on src/main/java/EchoBot.java:
```diff
@@ -0,0 +1,154 @@
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
```
Maybe you can move your EchoBot class under the duke package considering the flow.

newway1814 commented on src/main/java/EchoBot.java:
```diff
@@ -0,0 +1,154 @@
+import duke.command.UnmarkCommand;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.ui.Ui;
```
I think since you have changed Duke to echobot , you might consider renaming the package to EchoBot.


All in all, I think your codebase is very clean as I could easily track your code and never have to search up for the program flow. 👍

newway1814 replied:
I think since you have changed Duke to echobot , you might consider renaming the package to EchoBot.

All in all, I think your codebase is very clean as I could easily track your code and never have to search up for the program flow. 👍

newway1814 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,158 @@
+     * @return Task object.
+     */
+    public static Task fromFileString(String fileString) {
+        String[] parts = fileString.split("\\s*\\|\\s*"); // Adjusted delimiter
```
I liked how you used the split method here.

newway1814 commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,61 @@
+
+            // Additional information
+            if (task instanceof Event) {
+                LocalDateTime from = ((Event) task).getFrom();
```
Maybe you can write a javadoc for why are you typecasting to Event.

newway1814 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,100 @@
+        } catch (IOException e) {
+            chatArea.append("An error occurred while handling file operations: " + e.getMessage() + "\n");
+        } catch (Exception e) {
+            Logger logger = Logger.getLogger(Storage.class.getName());
```
Liked your implementation of logger here.

Daphne789 committed:
Modify GUI Implementation

Daphne789 committed:
Complete A-Assertions

Daphne789 committed:
Implement D-Notes feature: Add and List

Daphne789 committed:
Fix Checkstyle Issues

I ran the Checkstyle tool in the Gradle project and made necessary changes.

Daphne789 committed:
Complete A-CodeQuality

Let's,
* Fix Checkstyle issues to adhere to coding standards and conventions
* Add JavaDoc comments to provide clear documentation for classes and methods
* Improve code readability by applying abstraction where appropriate

Daphne789 committed:
Modify Level-10

Daphne789 committed:
Modify A-CodeQuality

Daphne789 committed:
Add Assertions

Daphne789 committed:
Merge pull request #2 from Daphne789/branch-A-Assertions

Add Assertions

Daphne789 committed:
Refactor long methods

SLAP is important to improve the code readability.

Let's,
- break down the long method into several smaller methods,
  including start(), handleUserInput() in EchoBot.java
  and loadTasks(), loadNotes() in Storage.java

Daphne789 committed:
Add Delete Note Feature

Daphne789 committed:
Add Ui.png

Daphne789 committed:
Add a User Guide

Daphne789 committed:
Fix the test to properly testing EchoBot

Daphne789 committed:
Modify the User Guide and fix check style

Daphne789 committed:
Modify the build.gradle and Change the Ui.png

Daphne789 committed:
Update README.md
